server/routes/basicDetails.ts

```diff
@@ -1,38 +1,44 @@
-import { type RequestHandler, Response, Router } from 'express'
+import { type Response, Router } from 'express'
 import { LocalDate, LocalDateTime } from '@js-joda/core'
-import asyncMiddleware from '../middleware/asyncMiddleware'
-import type { Services } from '../services'
 import AuditService, { Page } from '../services/auditService'
 import BreachNoticeApiClient, {
-  Address,
-  AddressList,
-  BasicDetails,
   BreachNotice,
   ErrorMessages,
-  Name,
   RadioButton,
   SelectItem,
 } from '../data/breachNoticeApiClient'
+import NdeliusIntegrationApiClient, { Address, AddressList, BasicDetails } from '../data/ndeliusIntegrationApiClient'
 import { fromUserDate, toUserDate } from '../utils/dateUtils'
 import { HmppsAuthClient } from '../data'
+import CommonUtils from '../services/commonUtils'
 
 export default function basicDetailsRoutes(
   router: Router,
   auditService: AuditService,
   hmppsAuthClient: HmppsAuthClient,
+  commonUtils: CommonUtils,
 ): Router {
   // const router = Router()
   // const get = (path: string | string[], handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
   // const post = (path: string | string[], handler: RequestHandler) => router.post(path, asyncMiddleware(handler))
 
+  const currentPage = 'basic-details'
+
   router.get('/basic-details/:id', async (req, res, next) => {
     await auditService.logPageView(Page.BASIC_DETAILS, { who: res.locals.user.username, correlationId: req.id })
     const breachNoticeApiClient = new BreachNoticeApiClient(
       await hmppsAuthClient.getSystemClientToken(res.locals.user.username),
     )
+    const ndeliusIntegrationApiClient = new NdeliusIntegrationApiClient(
+      await hmppsAuthClient.getSystemClientToken(res.locals.user.username),
+    )
     const { id } = req.params
-    const basicDetails = createDummyBasicDetails()
     const breachNotice = await breachNoticeApiClient.getBreachNoticeById(id)
+    const redirect = await commonUtils.redirectOnStatusChange(breachNotice, res)
+    if (redirect) {
+      return
+    }
```

Comment:
Does `redirectOnStatusChange` need to be async?  

Also, would it be clearer to rename this to something like `isCompleted(...)`. Then you could do something like:
```suggestion
    if (commonUtils.isCompleted(breachNotice, res)) return
```
and not worry about whether it's redirecting, just know not to continue if the breach notice has completed?

Reply:
Its going to include LAO check too. One check for changes which is why it was genericised.

Reply:
Ah I see that's why it's asynchronous - for the LAO API call.  Still, might be worth making it a one-liner if we'll have to repeat it in a few places?

E.g.
```suggestion
    if (await commonUtils.redirectRequired(breachNotice, req, res)) return
```